publication/models/account_analytic_invoice_line.py

```diff
@@ -28,3 +28,55 @@ def action_distribution_list(self):
         action['view_mode'] = 'form'
         action['target'] = 'current'
         return action
+
+    @api.model
+    def create(self, vals):
+        result = super(AccountAnalyticInvoiceLine, self).create(vals)
+        if result.publication:
+            self._create_publication_list()
+        return result
+
+    @api.multi
+    def write(self, vals):
+        result = super(AccountAnalyticInvoiceLine, self).write(vals)
+        self.filtered(
+            lambda x: x.publication is True)._create_publication_list()
+        return result
+
+    @api.multi
+    def _create_publication_list(self):
+        """ Conditionally creates a publication list:
+        If a list with the same contract_partner_id, product_id and
+        partner_id exists go write on that, else create a new one.
+        """
+        d_list = self.env['distribution.list']
+        for rec in self:
+            pub_list = d_list.search([
+                ('contract_partner_id',
+                    '=',
+                    rec.analytic_account_id.partner_id.id),
+                ('product_id', '=', rec.product_id.id),
+                ('partner_id', '=', rec.partner_id.id),
+            ], limit=1)
+            if pub_list:
+                pub_list.write({'copies': pub_list.copies + rec.quantity})
+            else:
+                d_list.create({
+                    'contract_partner_id':
+                        rec.analytic_account_id.partner_id.id,
+                    'product_id': rec.product_id.id,
+                    'partner_id': rec.partner_id.id,
+                })
+
+    @api.multi
+    def unlink(self):
+        records = self.filtered(lambda x: x.publication is True)
+        for rec in records:
+            self.env['distribution.list'].search([
```

Comment:
And you did not take into account the possibility that there might still be another contract.

Reply:
@NL66278 Not sure what you want me to do here.
Checking my notes on our discussion I see that:

> if we unlink a link of a publication product, find all the distribution lists that ref
> er to that product and receiving partner and delete them                             

Since the `distribution.list` records are not connected to contracts in any way, how can I find that?

Reply:
NB: they have to be deleted if it is the LAST contract line for this combination of (contract) partner and product. Mostly it will be the ONLY. If there are still contract lines for partner/product left: update distribution list same way as with a decrease of the quantity in the contract line. 

And of course you can find the distribution list entries searching on contract_partner_id and product_id.